internal/wasi/wasi_test.go

```diff
@@ -1147,11 +1147,19 @@ func TestSnapshotPreview1_FdWrite_Errors(t *testing.T) {
 		}
 	})
 
+	// Setup valid test memory
+	iovs, iovsCount := uint64(0), uint64(1)
```

Comment:
so basically here, we aren't truncating manually anymore, rather using the same memory for the "read side" of a 2 step operation like fdwrite. Ack it is easy to goof the offsets as the errno is less specific than normal go code where you'd use an `error` which would have some description of what context failed (ex reading offset vs len)